public/app/features/explore/TraceView/components/types/links.ts

```diff
@@ -4,18 +4,20 @@ import { Field } from '@grafana/data';
 
 import { TraceSpan } from './trace';
 
+export enum SpanLinkType {
+  Logs = 'log',
+  Traces = 'trace',
+  Metrics = 'metric',
+  Unknown = 'unknown',
+}
+
 export type SpanLinkDef = {
   href: string;
   onClick?: (event: unknown) => void;
   content: React.ReactNode;
   title?: string;
   field: Field;
+  type: SpanLinkType;
```

Comment:
@aocenas I ended up keeping a type here because there is a button that is just for logs, I figured we'd want to keep that working.